messages/teleporter/message_handler.go

```diff
@@ -65,10 +72,18 @@ func NewMessageHandlerFactory(
 		return nil, err
 	}
 
+	var deciderClient pbDecider.DeciderServiceClient
+	if grpcClient == nil {
+		deciderClient = nil
```

Comment:
Setting an interface value to nil that will subsequently be called gives me the heebie-jeebies. Could we implement a struct `emptyDeciderServiceClient` that just returns true when `ShouldSendMessage` is called?

Reply:
still new to Golang here, so thanks for bearing with me. a little research suggests that `nil` is the zero (uninitialized) value for interfaces, so this shouldn't matter, right? in any case, since this assignment is redundant anyways, I took it out in https://github.com/ava-labs/awm-relayer/pull/344/commits/682fead651913a9142ff49c59c52844b3a8b66b3.

is the "empty struct" thing a common idiom? I'm happy to defer to convention, but I'd appreciate a little reading material to help me understand the reasoning behind the idiom.

Reply:
I took your suggestion about the empty struct thing in https://github.com/ava-labs/awm-relayer/pull/344/commits/f8662fdaf33fa9d196655df3ef485144d089e69d